### PR TITLE
Complete prep-javascript-loops - Nader Motamedi

### PR DESCRIPTION
**Pull request by nmotamedi**
Screen Shot 2024-01-13 at 12 08 14 PM
Screen Shot 2024-01-13 at 12 08 03 PM

### CODE REVIEW

File: prep-javascript-loops/main.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
+// create your loops here.
+function whileLoop1(){
+  const newArray = [];
+  n = 0;
```
Comment (shawnkost) [inline prep-javascript-loops/main.js:4]: Don't forget your variable initializer! There should be a `let` here

File: prep-javascript-loops/main.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    newArray.push(n);
+    n++;
+  }
+  return newArray;
+}
+console.log("whileLoop1 output", whileLoop1())
+
+function whileLoop2(){
+  const newArray = [];
+  n=0;
```
Comment (shawnkost) [inline prep-javascript-loops/main.js:15]: Don't forget your variable initializer! There should be a `let` here

File: prep-javascript-loops/prep-javascript-loops-quiz-notes.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+- When is the first expression in the parentheses for a `for` loop (known as the **initialization**) evaluated?
+
+- When is the second expression in the parentheses for a `for` loop (known as the **condition**) evaluated?
+
+- When is the third expression in the parentheses for a `for` loop (known as the **final expression**) evaluated?
+
+- What is the purpose of the **final expression** in a `for` loop?
+
+- What types of data should the `for...in` loop be used on?
+
```
Comment (shawnkost) [inline prep-javascript-loops/prep-javascript-loops-quiz-notes.md:18]: Don't forget about the quiz questions!

**Review by shawnkost — CHANGES_REQUESTED** [review]
Looking good for the most part Nader! Just mainly missing the quiz questions

**Comment by nmotamedi** [comment]
Screen Shot 2024-01-13 at 12 29 23 PM
Screen Shot 2024-01-13 at 12 29 16 PM
Screen Shot 2024-01-13 at 12 29 10 PM

**Review by shawnkost — APPROVED** [review]
👍🏻

**Pull request merged**